Source/RapyutaSimulationPlugins/Public/Robots/RRRobotROS2Interface.h

```diff
@@ -0,0 +1,123 @@
+/**
+ * @file RRRobotROS2Interface.h
+ * @brief Base Robot ROS controller class. Other robot ROS2 class should inherit from this class.
+ * @copyright Copyright 2020-2022 Rapyuta Robotics Co., Ltd.
+ */
+
+#pragma once
+
+// UE
+#include "CoreMinimal.h"
+
+// rclUE
+#include "ROS2Node.h"
+#include "Tools/RRROS2OdomPublisher.h"
+
+#include "RRRobotROS2Interface.generated.h"
+
+class ARobotVehicle;
+/**
+ * @brief  Base Robot ROS2 interface class.
+ * This class owns ROS2Node and provide ROS2 interfaces to control robot such as Twist msg.
+ */
+UCLASS(Blueprintable)
+class RAPYUTASIMULATIONPLUGINS_API URRRobotROS2Interface : public UObject
+{
+    GENERATED_BODY()
+
+#define RR_ROBOT_ROS2_SUBSCRIBE_TO_TOPIC(InTopicName, InMsgClass, InCallback) \
+    RR_ROS2_SUBSCRIBE_TO_TOPIC(RobotROS2Node, this, InTopicName, InMsgClass, InCallback)
+
+public:
+    //! Target robot
+    UPROPERTY(Transient)
+    ARobotVehicle* Robot = nullptr;
+
+    //! Target ROS2 node of this interface
+    UPROPERTY(Transient)
+    AROS2Node* RobotROS2Node = nullptr;
+
+    /**
+     * @brief Initialize robot's ROS2 interface
+     *
+     * @param InRobot
+     */
+    virtual void Initialize(ARobotVehicle* InRobot);
+
+    /**
+     * @brief Spawn ROS2Node and initialize it. This method is mainly used by #ASimulationState to spawn from ROS2 service.
+     *
+     * @param InPawn
+     */
+    void InitRobotROS2Node(ARobotVehicle* InRobot);
+
+    UPROPERTY(Transient, BlueprintReadWrite)
+    URRROS2OdomPublisher* OdomPublisher = nullptr;
+
+    /**
+     * @brief Initialize non sensor publishes such as odometry.
+     *
+     */
+    UFUNCTION()
+    virtual bool InitPublishers();
+
+    /**
+     * @brief Stop all publishers
+     *
+     */
+    UFUNCTION()
+    virtual void StopPublishers();
+
+    /**
+     * @brief Initialize subscriptions
+     *
+     */
+    UFUNCTION()

```

Comment:
better to have more explanation for init subscripsions, e.g. 
- this method is expected to be override for each robot
- base class has cmd_vel and joint state subscription